src/crimson/os/seastore/omap_manager/btree/omap_btree_node_impl.h

```diff
@@ -156,10 +262,51 @@ using OMapInnerNodeRef = OMapInnerNode::OMapInnerNodeRef;
 
 struct OMapLeafNode
   : OMapNode,
-    StringKVLeafNodeLayout {
-
+    StringKVLeafNodeLayout,
+    ChildNode<OMapInnerNode, OMapLeafNode, std::string> {
   using OMapLeafNodeRef = TCachedExtentRef<OMapLeafNode>;
-  using internal_iterator_t = const_iterator;
+  using internal_const_iterator_t = const_iterator;
+  using base_child_t = BaseChildNode<OMapInnerNode, std::string>;
+  using child_node_t = ChildNode<OMapInnerNode, OMapLeafNode, std::string>;
+
+  void do_on_rewrite(Transaction &t, LogicalCachedExtent &extent) final {
+    auto &other = static_cast<OMapInnerNode&>(extent);
+    this->init_range(other.get_begin(), other.get_end());
+  }
+
+  void on_invalidated(Transaction &t) final {
+    this->child_node_t::on_invalidated();
+  }
+
+  void prepare_commit() final {
+    if (is_rewrite() && !is_btree_root()) {
+      auto &prior = *get_prior_instance()->template cast<OMapLeafNode>();
+      if (prior.base_child_t::has_parent_tracker()) {
```

Comment:
Same here, looks we can assume that `prior` must always has parent-tracker if it isn't the root ?

Reply:
Um, not really in this case. The omap node being rewritten might be loaded by the journal replaying, which means it may not be linked to its parent.

Reply:
related to https://github.com/ceph/ceph/pull/61936#discussion_r2041477805